### DISCUSSION TITLE
Adding i18n to custom navigation component

### DISCUSSION THREAD

**Discussion by gabrielgomeso** [vuejs/vitepress#4150] [Q&A] [answered]
I've managed to create a custom component and add it to my theme (created a `/theme/components/TheHeader.vue` file and declared the component inside `theme/index.ts` enhanceApp).
I'm having some trouble adding i18n to the navigation inside this header - I have no idea how to do it. I was able to add i18n to the default theme, which was quite easy (there is [this discussion](https://github.com/vuejs/vitepress/discussions/1839) about it), but I cannot find in the docs or at GitHub anyone that attempted to create a custom header/navigation and add i18n to it
Any tips on how to do that?

**Comment by brc-dd** [comment, marked as answer]
Just use 

```ts
import { useData } from 'vitepress'
import type YourThemeConfig from './some/path'

const { theme } = useData<YourThemeConfig>() // <-- theme is read-only ref here. You just need to render theme.value.nav or something in your header and vue will track the rest. 
```
